### PR TITLE
Esa ehst download improvements

### PR DESCRIPTION
**Pull request by jespinosaar**
Dear Astroquery Team,

This PR aims to offer a more intuitive way for users to download products from eHST Science Archive. I have left get_artifact method to ensure backwards compatibility, but I have moved its functionality to download_file method. In addition to this, we have added a mechanism to retrieve the files associated to an observation and download only FITS files. Finally, the criteria methods have been updated so now users can search by proposal.

Let's iterate in this new pull request. Please let me know your thoughts and comments. Thanks in advance for your support!

cc @esdc-esac-esa-int

### CODE REVIEW

**Review by keflavich — COMMENTED** [review]
Looks good to me.  The new approach seems simpler.  It might be helpful to note in the documentation that the expected file extension has changed.

I'm not going to mark this as "Approve", though, as I think someone with more direct experience should review if possible.

File: astroquery/esa/hubble/tests/test_esa_hubble_remote.py
Comment (bsipocz) [inline astroquery/esa/hubble/tests/test_esa_hubble_remote.py:62]: I'm afraid this won't work with windows (we still have a bit of a cleanup backlog for that, including some in the esa modules https://github.com/astropy/astroquery/issues/2773). So while we're at it, could you swap all these string manipulations with an `os.path.join` instead, here and below, too?

File: docs/esa/hubble/hubble.rst
Comment (bsipocz) [inline docs/esa/hubble/hubble.rst:380]: Add a note or some other directive around this to highlight.

File: docs/esa/hubble/hubble.rst
Comment (bsipocz) [inline docs/esa/hubble/hubble.rst:391]: Although we're ignoring the outputs (but only do that if frequent changes are expected, e.g. more observations are coming in, the order of the result is not deterministic, etc), please include a few line long of the expected outcome. E.g. in this case, as I see it will be only the string `raw_data_for_j6fl25s4q.zip`

OTOH, all these download examples are actually downloading the files, so I wonder whether cutting back on the actual code examples would make sense (after all your narrative text does the explanation quite well)

**Review by bsipocz — COMMENTED** [review]
I have a few minor comments, otherwise this is almost ready to go. Thanks!

(ignore the failing devtest job, we're dealing with it separately and it won't hold up this PR)

File: astroquery/esa/hubble/core.py
Comment (jespinosaar) [reply astroquery/esa/hubble/core.py:53]: This is the main issue of copy-pasting ;)

File: astroquery/esa/hubble/core.py
Comment (jespinosaar) [reply astroquery/esa/hubble/core.py]: Done!

File: CHANGES.rst
Comment (jespinosaar) [reply CHANGES.rst]: Thanks for the suggestion, included.

File: astroquery/esa/hubble/core.py
Comment (jespinosaar) [reply astroquery/esa/hubble/core.py]: Reviewed and updated, the sentence is complete now.

File: CHANGES.rst
Comment (jespinosaar) [reply CHANGES.rst]: I included your suggestion, thanks!

File: astroquery/esa/hubble/tests/test_esa_hubble_remote.py
Comment (jespinosaar) [reply astroquery/esa/hubble/tests/test_esa_hubble_remote.py]: Modified in different places so it works on every environment.

File: docs/esa/hubble/hubble.rst
Comment (jespinosaar) [reply docs/esa/hubble/hubble.rst]: Notes and warnings included.

File: docs/esa/hubble/hubble.rst
Comment (jespinosaar) [reply docs/esa/hubble/hubble.rst]: I added the expected output. Thanks!

File: astroquery/esa/hubble/core.py
Comment (bsipocz) [inline astroquery/esa/hubble/core.py:289]: Same API considerations, maybe this could be called `download_files`, and having a kwarg for extensions defaulting to `'fits'`? This is really only a discussion item as `download_files` is not really wide-spread, and the currently existing ones in alma and casda are having different signature

**Review by bsipocz — APPROVED** [review]
One minor namespace fix that I commit directly, otherwise it's good to go. Thanks @jespinosaar. 

However, with the new added methods I did a quick check on the current API, and saw that they may benefit from naming them slightly differently to add some consistency. I would suggest to consider these comments, and address it in a follow-up?

**Comment by bsipocz** [comment]
Thanks @jespinosaar! About the method names I opened a reminder issue as it doesn't need to hold up the merge of this one.

**Comment by jespinosaar** [comment]
Many thanks @bsipocz and @keflavich! Yes, we can fix the method names in other PR.

File: astroquery/esa/hubble/core.py
Comment (jespinosaar) [reply astroquery/esa/hubble/core.py:289]: We wanted a specific method naming FITS, to make it more visible to users.

File: astroquery/esa/hubble/core.py
Comment (jespinosaar) [reply astroquery/esa/hubble/core.py:265]: Not the same, in this case we are only listing the file names, in case users want to filter them before download anything. get_obs_products downloads them.

**Pull request merged**
